workers/unity/Assets/Fps/Scripts/GameLogic/WorldTiles/MapQualitySettings.cs

```diff
@@ -0,0 +1,136 @@
+﻿﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Fps
+{
+    [CreateAssetMenu(fileName = "MapQualitySettings", menuName = "Improbable/Map Quality Settings")]
+    public class MapQualitySettings : ScriptableObject
+    {
+        public const float DefaultCheckoutDistance = 60f;
+
+        public static bool ShowPreview;
+        public List<MapQualityLevelData> Settings = new List<MapQualityLevelData>();
+
+        private static MapQualitySettings instance;
+        private float checkoutDistanceCache = -1;
+
+        public static float CheckoutDistance => Instance.CheckoutDistanceInternal;
+
+        public static MapQualitySettings Instance
+        {
+            get
+            {
+                if (instance == null)
+                {
+                    CreateDefaultInstance();
+                }
+
+                return instance;
+            }
+
+            set
+            {
+                if (value == null)
+                {
+                    CreateDefaultInstance();
+                }
+                else
+                {
+                    instance = value;
+                    instance.Apply();
+                }
+            }
+        }
+
+        private static void CreateDefaultInstance()
+        {
+            Debug.LogWarning("MapQualitySettings: No settings specified. To fix this, use a " +
+                "MapQualitySettingsPublisher and ensure it has a MapQualitySettings reference.\n");
+
+            instance = CreateInstance<MapQualitySettings>();
+            instance.Apply();
+        }
+
+        private float CheckoutDistanceInternal
+        {
+            get
```

Comment:
This is a really big property and does a fair bit. Would prefer is this was an explicit function. I.e. 

```private float CalculateCheckoutDistance()```

Reply:
since the property `CheckoutDistance` just calls `Instance.CheckoutDistanceInternal`, should `CheckoutDistance` itself become a function (like `GetCheckoutDistance()`)?

Reply:
Turned into a function, but `static float CheckoutDistance` is still a property